src/views/Blog.vue

```diff
@@ -32,19 +32,21 @@
 <script>
 import { getTagsArray } from '@/utility/tags';
 import get from 'lodash/get';
-import { getBlog } from '@/api/blogService.js';
+import { getBlog, addViewOnBlog } from '@/api/blogService';
 
 export default {
   name: 'Blog',
   data() {
     return {
+      blogId: this.$route.params.blogId,
       blog: null,
     };
   },
   async mounted() {
     try {
-      let data = await getBlog(this.$route.params.blogId);
+      let data = await getBlog(this.blogId);
```

Comment:
I suggest to move all this code in a method.
In mounted you will just call that method